### PR TITLE
Feature-2 Models

### PR DESCRIPTION
**Pull request by ZAINABFATIMA0**
Added event, schedule and category models in events app
Added creator and registrations models in users app
Added email_log models in communication app

### CODE REVIEW

File: communication/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class CommunicationConfig(AppConfig):
+    default_auto_field = "django.db.models.BigAutoField"
+    name = "communication"
```
Comment (sheralim012) [inline communication/apps.py]: app name is usually plural

File: communication/models.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+from users.models import Registrations
+
+
+class Email_log(models.Model) :
```
Comment (sheralim012) [inline communication/models.py]: seriously, underscore in class name, not good

File: communication/models.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    email_type = models.CharField (max_length=255)
+    time_stamp = models.DateTimeField(auto_now_add=True)
```
Comment (sheralim012) [inline communication/models.py]: timestamp is a single word

File: users/models.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+class User(AbstractUser):
+    email = models.EmailField(max_length=255, unique=True)
+    phone = models.CharField(max_length=32, unique=True, blank=True, null=True)
+
+    USERNAME_FIELD = "email"
+
+
+class Registrations(models.Model):
+    email = models.EmailField(max_length=200, unique=True)
```
Comment (sheralim012) [inline users/models.py]: should this be unique, if someone register in one event, he/she wouldn't be able to register in an other

File: events/models.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    event_type = models.CharField(
+        max_length=20,
+        choices=EVENT_TYPE_CHOICES,
+        default=ONSITE,
+    )
+    registration_start_time = models.DateTimeField()
+    registration_end_time = models.DateTimeField()
+    meeting_link = models.URLField()
+    seat_limit = models.PositiveIntegerField()
+    unique_link = models.URLField(unique=True)
```
Comment (sheralim012) [inline events/models.py]: what is this for?

**Review by sheralim012 — CHANGES_REQUESTED** [review]
- a lot of formatting issues
- no requirements file
- no env file
- repeated mistakes

make sure to review your own PR before sending it to me

File: events/models.py (hunk 5 above)
Comment (ZAINABFATIMA0) [reply events/models.py]: I added this field to store the url specific for every event. Should i do it like this or should i just use the event's id ?

File: communications/migrations/0001_initial.py
Comment (sheralim012) [inline communications/migrations/0001_initial.py:1]: per PR always minimize the number of migration file

File: events/models.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        "events.Category", 
+        related_name="categories", 
+        null=True, 
+        blank=True, 
+        on_delete=models.CASCADE
+    )
+
+
+class Event(models.Model):
+    ONSITE = 'On-site'
```
Comment (sheralim012) [inline events/models.py]: still not update, read about django text choice

File: events/models.py (hunk 5 above)
Comment (sheralim012) [reply events/models.py]: shouldn't this be auto generated and not editable by user?

**Review by sheralim012 — CHANGES_REQUESTED** [review]

File: communications/migrations/0001_initial.py
Comment (ZAINABFATIMA0) [reply communications/migrations/0001_initial.py:1]: These are the minimum numbers of migrations right now. It automatically creates two migration files for some apps due to their dependencies.

File: events/models.py (hunk 5 above)
Comment (ZAINABFATIMA0) [reply events/models.py]: Yes, you are right. I have removed this attribute.

File: communications/models.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
+from django.db import models
+
+
+class EmailLog(models.Model):
+    email_type = models.CharField(max_length=255)
```
Comment (sheralim012) [inline communications/models.py]: always avoid unnecessary prefixes,
as model name already has email in it then no need to mention it again

Comment (sheralim012) [reply communications/models.py]: model fields should be in alphabetic order
and should be divided in at least 2 section
first fields then related fields aka keys

File: communications/models.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    subject = models.CharField(max_length=255)
+    body = models.CharField(max_length=255)
+
+    timestamp = models.DateTimeField(auto_now_add=True)
+
+    recipient = models.ForeignKey(
+        "users.Registrations", 
+        related_name="email_log",
```
Comment (sheralim012) [inline communications/models.py]:
```suggestion
        related_name="email_logs", 
```

usually reverse names are plural `registration.email_logs`

File: events/models.py [hunk 9]
```diff
@@ -0,0 +1,62 @@
+from django.db import models
+from django.contrib.gis.db import models as model
+
+
+class Category(models.Model):
+    category_name = models.CharField(max_length=255)
```
Comment (sheralim012) [inline events/models.py]: same as above

File: events/models.py [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        related_name="categories", 
+        null=True, 
+        blank=True, 
+        on_delete=models.CASCADE
+    )
+
+
+class Event(models.Model):
+    class EventType(models.TextChoices):
+        onsite = 'ONSITE', 'Onsite'
```
Comment (sheralim012) [inline events/models.py]:
```suggestion
        ONSITE = 'ONSITE', 'Onsite'
```

File: events/models.py [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    event_type = models.CharField(
+        max_length=10,
+        choices=EventType.choices,
+        default=EventType.onsite,
+    )
+    name = models.CharField(max_length=32)
+
+    registration_start_time = models.DateTimeField()
+    registration_end_time = models.DateTimeField()
+
```
Comment (sheralim012) [inline events/models.py]: why so many empty lines

**Review by sheralim012 — CHANGES_REQUESTED** [review]
requirements file is still missing

File: events/models.py (hunk 11 above)
Comment (ZAINABFATIMA0) [reply events/models.py]: I did this to separate model fields and divide them in sections with same field for better readability purpose.

**Pull request merged**